opengrok-tools/src/main/python/opengrok_tools/deploy.py

```diff
@@ -27,77 +27,135 @@
 import tempfile
 from shutil import copyfile
 from zipfile import ZipFile
+import xml.etree.ElementTree as ET
 
 from .utils.log import get_console_logger, get_class_basename, \
     fatal
 from .utils.parsers import get_baseparser
 
+WEB_XML = 'WEB-INF/web.xml'
+DEFAULT_CONFIG_FILE = '/var/opengrok/etc/configuration.xml'
+
 """
  deploy war file
 
 """
 
 
-def repack_war(logger, sourceWar, targetWar, configFile, defaultConfigFile):
+class ProcessingException(Exception):
+    pass
+
+
+def insert_file(input_xml, insert_xml_file):
     """
-    Repack sourceWar into targetWar, performing substitution of configFile
-    in the process.
+    inserts sub-root elements of XML file under root of input XML
+    :param input_xml: input XML string
+    :param insert_xml_file: path to file to insert
+    :return: string with resulting XML
     """
+    # This avoids resulting XML to have namespace prefixes in elements.
+    ET.register_namespace('', "http://xmlns.jcp.org/xml/ns/javaee")
+
+    root = ET.fromstring(input_xml)
+    insert_tree = ET.parse(insert_xml_file)
+    insert_root = insert_tree.getroot()
+    index = len(root)
 
-    WEB_XML = 'WEB-INF/web.xml'
+    for elem in list(insert_root.findall('.')):
+        for e in list(elem):
+            root.insert(index, e)
```

Comment:
there must be something like append, no?

Reply:
good call, there is `extend()`: https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element.extend